src/main/java/net/imglib2/RandomAccess.java

```diff
@@ -53,4 +54,116 @@
 	RandomAccess< T > copyRandomAccess();
 //	@Override
 //	RandomAccess< T > copy();
+
+	/**
+	 * Convenience method that moves the {@link RealRandomAccess} to the given
+	 * position and gets the value at that position. It's a shortcut for:
+	 * <p>
+	 * <pre>
+	 * {@code
+	 * setPosition( position );
+	 * get();
+	 * }
+	 * </pre>
+	 * <p>
+	 * WARNING: The return value is invalidated by next call to
+	 * {@link #setPositionAndGet}  or {@link #setPosition}.
+	 * <p>
+	 * <pre>
+	 * {@code
+	 * // This is wrong!!!
+	 * a = randomAccess.setPositionAndGet( positionA );
+	 * b = randomAccess.setPositionAndGet( positionB ); // this invalidates "a" !!!
+	 * wrongDifference = a.getRealDouble() - b.getRealDouble();
+	 *
+	 * // Correct:
+	 * // Use individual RandomAccesses to query a and b
+	 * a = randomAccess_A.setPositionAndGet( positionA );
+	 * b = randomAccess_B.setPositionAndGet( positionB ); // this is fine because a different RandomAccess is used
+	 * difference = a.getRealDouble() - b.getRealDouble();
+	 * }
+	 * </pre>
+	 */
+	default T setPositionAndGet( long... position )
```

Comment:
@hanslovsky I didn't like the javadoc like it was before. It said, that this method has performance issues, that's just not the case. But at the same time I was missing the Warning, that you can see now, which I think is very important.

Reply:
Your are right `ListRandomAccess` is an exception. And also that `setPositionAndGet(...)` is just a shortcut, and exactly the same as two consecutive calls to `setPosition` and `get`. But still, I have a strong gut feeling that it will invite new imglib2 users to use it wrongly, and that the warning will be very useful.